src/webgpu/shader/validation/parse/loop.spec.ts

```diff
@@ -0,0 +1,59 @@
+export const description = `Validation parser tests for 'loop' statements`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kTests = {
+  break: { wgsl: `loop { break; }`, pass: true },
+  return: { wgsl: `loop { return; }`, pass: true },
+  break_continuing: { wgsl: `loop { break; continuing {} }`, pass: true },
+  var_break: { wgsl: `loop { var a = 1; break; }`, pass: true },
+  var_break_continuing_inc: {
+    wgsl: `loop { var a = 1; break; continuing { a += 1; }}`,
+    pass: true,
+  },
+  var_break_continuing_discard: {
+    wgsl: `loop { var a = 1; break; continuing { discard; }}`,
+    pass: true,
+  },
+
+  continue: { wgsl: `loop { continue; }`, pass: false },
```

Comment:
Worth noting in a comment that these cases fail not because of parsing but because of behavioural analysis. (also behaviour analysis is tested thoroughly in another spot.)

Reply:
Done